### PR TITLE
Amazon Kinesis Data Firehose input documentation

### LINKED ISSUE #2797: [REQUEST]: Add documentation in the observability guide for sending logs to Elasticsearch with AWS Kinesis Firehose
### Description

This is a new product feature within Infrastructure Observability for shipping logs to ESS via AWS Kinesis Firehose delivery streams. 

This solution is designed to be used alongside existing AWS integrations in Kibana (e.g. https://docs.elastic.co/en/integrations/aws/vpcflow), but can also be used for shipping any AWS logs. 

We will produce a setup guide, troubleshooting guide, and something more dry with reference configuration etc. 

### Resources

There are some draft documents but they are currently in internal Google docs/private GH issues, so not linking them in this public issue. Please reach out to me and I can provide the links. We will be creating a draft PR with the content shortly anyway. 

### Collaboration

The product team will provide the initial content.

### Point of contact.

**Main contact:** @tommyers-elastic @vinaychandrasekhar 

**Stakeholders:**

### PR DESCRIPTION
**Pull request by tommyers-elastic**
Closes #2797

### CODE REVIEW

**Review by girodav — COMMENTED** [review]
Great job :)! I just noticed that we do not mention what to put in "backup settings". See screenshot:

Screenshot 2023-03-29 at 09 53 47

As done for the other sections, it might be worth adding some lines about what the user can/should configure.

File: docs/en/observability/aws-firehose/aws-firehose-setup.asciidoc
Comment (tommyers-elastic) [reply docs/en/observability/aws-firehose/aws-firehose-setup.asciidoc:29]: thanks ! done

File: docs/en/observability/aws-firehose/aws-firehose-setup.asciidoc
Comment (bmorelli25) [inline docs/en/observability/aws-firehose/aws-firehose-setup.asciidoc:46]:
```suggestion
[[aws-firehose-create-delivery-stream]] 
. Create a delivery stream in Amazon Kinesis Data Firehose.
```

File: docs/en/observability/aws-firehose/aws-firehose-setup.asciidoc
Comment (bmorelli25) [inline docs/en/observability/aws-firehose/aws-firehose-setup.asciidoc:61]: This should be aws-firehose-...
```suggestion
Unless you are streaming data from Kinesis Data Streams, set source to *Direct PUT* (see <<aws-firehose-send-data-to-delivery-stream>> for more details on data sources).
```

File: docs/en/observability/aws-firehose/aws-firehose-setup.asciidoc
Comment (bmorelli25) [inline docs/en/observability/aws-firehose/aws-firehose-setup.asciidoc:193]: anchor needs to be above the list, and `h` instead of `b`
```suggestion
[[aws-firehose-send-data-to-delivery-stream]] 
. Send data to the Firehose delivery stream.
```

File: docs/en/observability/aws-firehose/aws-firehose.asciidoc
Comment (bmorelli25) [inline docs/en/observability/aws-firehose/aws-firehose.asciidoc:20]: All image paths need to be updated
```suggestion
image::./images/firehose-architecture.png[Diagram showing Amazon Kinesis Data Firehose connected to Elastic Cloud with examples of input data sources, such as Amazon Kinesis Data Streams and Amazon Route 53 logs.]
```

File: docs/en/observability/aws-firehose/aws-firehose-setup.asciidoc
Comment (bmorelli25) [inline docs/en/observability/aws-firehose/aws-firehose-setup.asciidoc:44]:
```suggestion
image::./images/firehose-integrations-install-assets.png[AWS integration settings page with the "Install AWS assets" button highlighted]
```

File: docs/en/observability/aws-firehose/aws-firehose-setup.asciidoc
Comment (bmorelli25) [inline docs/en/observability/aws-firehose/aws-firehose-setup.asciidoc:52]:
```suggestion
image::./images/firehose-create-delivery-stream.png[Amazon Kinesis dashboard with the "Create delivery stream" button highlighted]
```

File: docs/en/observability/aws-firehose/aws-firehose-setup.asciidoc
Comment (bmorelli25) [inline docs/en/observability/aws-firehose/aws-firehose-setup.asciidoc:81]:
```suggestion
image::./images/firehose-config-1.png[Amazon Kinesis Data Firehose delivery stream settings showing 'Choose source and destination', 'Delivery stream name' and 'Transform records' sections]
```

File: docs/en/observability/aws-firehose/aws-firehose-setup.asciidoc
Comment (bmorelli25) [inline docs/en/observability/aws-firehose/aws-firehose-setup.asciidoc:168]:
```suggestion
image::./images/firehose-integration-data-stream.png[AWS WAF integration documentation showing a sample event with the data stream components highlighted]
```

File: docs/en/observability/aws-firehose/aws-firehose-setup.asciidoc
Comment (bmorelli25) [inline docs/en/observability/aws-firehose/aws-firehose-setup.asciidoc:191]:
```suggestion
image::./images/firehose-config-2.png[Amazon Kinesis Data Firehose delivery stream settings showing 'Destination settings' section]
```

File: docs/en/observability/aws-firehose/aws-firehose-setup.asciidoc
Comment (bmorelli25) [inline docs/en/observability/aws-firehose/aws-firehose-setup.asciidoc:203]: This needs link text, otherwise the entire URL displays.

File: docs/en/observability/aws-firehose/aws-firehose-setup.asciidoc
Comment (bmorelli25) [inline docs/en/observability/aws-firehose/aws-firehose-setup.asciidoc:17]:
```suggestion
* When using https://www.elastic.co/integrations[Elastic integrations] with Firehose, only a single log type may be sent per delivery stream, e.g. VPC Flow Logs.
```

File: docs/en/observability/aws-firehose/aws-firehose-setup.asciidoc
Comment (bmorelli25) [inline docs/en/observability/aws-firehose/aws-firehose-setup.asciidoc:78]:
```suggestion
For advanced use cases, source records can be transformed by invoking a custom Lambda function.
When using Elastic integrations, this should not be required. 
```

File: docs/en/observability/aws-firehose/aws-firehose-setup.asciidoc
Comment (bmorelli25) [inline docs/en/observability/aws-firehose/aws-firehose-setup.asciidoc:123]: We can almost always replace "make use of" with "use"

```suggestion
Elastic integrations use data streams with specific naming conventions, and Firehose records need to be routed to the relevant data stream to use preconfigured index mappings, ingest pipelines, and dashboards.
```

File: docs/en/observability/aws-firehose/aws-firehose-setup.asciidoc
Comment (bmorelli25) [inline docs/en/observability/aws-firehose/aws-firehose-setup.asciidoc:202]: This link needs link text to render nicely

File: docs/en/observability/aws-firehose/aws-firehose.asciidoc
Comment (bmorelli25) [inline docs/en/observability/aws-firehose/aws-firehose.asciidoc:23]: Since this is technical preview, you could just add this part in later.

File: docs/en/observability/aws-firehose/aws-firehose-setup.asciidoc
Comment (bmorelli25) [inline docs/en/observability/aws-firehose/aws-firehose-setup.asciidoc:93]:
```suggestion
Set *Elastic endpoint URL* to point to your {es} cluster running in {ecloud}.
This endpoint can be found in the {ess-console}[{ecloud} console].
```

File: docs/en/observability/aws-firehose/aws-firehose-setup.asciidoc
Comment (bmorelli25) [inline docs/en/observability/aws-firehose/aws-firehose-setup.asciidoc:162]: I'd rather link to the documentation about data stream naming since that will continue to be updated:

```suggestion
As per the {fleet-guide}/data-streams.html[data stream naming conventions], the "namespace" is a user-configurable arbitrary grouping and can be changed from `default` to fit your use case. For example, you may want to organize WAF Logs per environment into `logs-aws.waf-production` and `logs-aws.waf-qa` data streams for more granular control over rollover, retention, and security permissions.
```

File: docs/en/observability/aws-firehose/aws-firehose-setup.asciidoc
Comment (bmorelli25) [inline docs/en/observability/aws-firehose/aws-firehose-setup.asciidoc:184]:
```suggestion
Elastic requires a *Buffer size* of 1MiB to avoid exceeding the {es} `http.max_content_length` setting (typically 100MB) when the buffer is uncompressed.
```

**Review by bmorelli25 — COMMENTED** [review]
Looking good 😄. I was unable to push to your branch, so I have quite a few recommendations.

**Comment by dedemorton** [comment]
@tommyers-elastic If you'd like, I can go through and apply Brandon's changes to get this moving forward. Just let me know.

Also...I wanted to mention that we are in the process of moving the ESF docs out of the Observability Guide and into a separate "book." We're doing this because ESF has its own versioning scheme. Would it make sense to do the same for the Amazon Kinesis Data Firehose docs (rather than putting them in the Obs Guide). Let me know what you think.

**Comment by dedemorton** [comment]
@tommyers-elastic I have a couple questions:

* Will these capabilities follow the same versioning scheme as the stack docs?
* Would you prefer to see this content grouped with other native cloud shippers (like ESF)? (There's a [PR in progress right now](https://github.com/elastic/docs/pull/2634) that moves the ESF docs to a separate (new) book to allow for non-stack versioning.)

Probably best to decide the location now so we don't have to set up redirects later. WDYT?

File: docs/en/observability/aws-firehose/aws-firehose.asciidoc
Comment (dedemorton) [inline docs/en/observability/aws-firehose/aws-firehose.asciidoc:28]:
```suggestion
//[[aws-firehose-comparison]]
//== Comparison with other ways to send data to Elastic

//TODO
```
Maybe just comment out for now?

File: docs/en/observability/aws-firehose/aws-firehose-setup.asciidoc
Comment (dedemorton) [inline docs/en/observability/aws-firehose/aws-firehose-setup.asciidoc:205]:
```suggestion
* Publish VPC Flow Logs to a Cloudwatch log group. To learn how, refer to the https://docs.aws.amazon.com/vpc/latest/userguide/flow-logs-cwl.html[AWS documentation about publishing flow logs].
```

File: docs/en/observability/aws-firehose/aws-firehose-setup.asciidoc
Comment (dedemorton) [inline docs/en/observability/aws-firehose/aws-firehose-setup.asciidoc:206]:
```suggestion
* Create a subscription filter in the CloudWatch log group to the Firehose delivery stream. To learn how, refer to the https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/SubscriptionFilters.html#FirehoseExample[AWS documentation about using subscription filters].
```

**Review by dedemorton — COMMENTED** [review]
Adds suggestions to resolve remaining comments.

**Review by dedemorton — COMMENTED** [review]
Add suggestions to resolve remaining feedback

**Comment by tommyers-elastic** [comment]
hey @dedemorton thanks for the review and help on this PR! 

regarding the location of the docs and versioning. this is not tied to stack release versions, but for under-the-hood reasons _is_ tied to ESS release milestones. in the future though this we be silently decoupled from the ESS releases. 

I think keeping it alongside the other native cloud shippers like ESF sounds like a good idea. what would the process be to do that? merge this PR and then port it there?

Thanks!

**Comment by dedemorton** [comment]
@tommyers-elastic It doesn't really matter where the source files live. I just need to tell the build where to find the files and which branches to use. So obviously the repo needs to have branches that match the versioning scheme you want to use.

We could start out with the source files in `main` in the `observability-docs` repo  if you don't have a separate repo for the code yet.

Also...I should mention that the ESF repo uses tags rather than branches to build versions, but we are planning to add branches to support the docs build.

File: docs/en/observability/aws-firehose/aws-firehose.asciidoc
Comment (dedemorton) [inline docs/en/observability/aws-firehose/aws-firehose.asciidoc:13]:
```suggestion

[discrete]
[[aws-firehose-what-is-it]]
```
Just add a discrete tag to keep this section on the same page.

Comment (dedemorton) [reply docs/en/observability/aws-firehose/aws-firehose.asciidoc:13]: I've added a suggestion that should fix the problem. I'll commit the suggestion so we can see the result,

**Comment by dedemorton** [comment]
@tommyers-elastic I'll be on PTO next week. If you'd like to get this merged and published, I recommend that you:

- Remove the lines where you include the firehose files in the observability docs [here](https://github.com/elastic/observability-docs/pull/2802/files#diff-684c7d57d45e7d7c685a504568d48f8e3988685b66c4391800b7c9f6c029db9eR69).
- Merge this PR.
- Update the PR I have open in the docs repo and add a section for the firehose docs. You can see how to do this [here](https://github.com/elastic/docs/pull/2634/files#diff-9925a85f7cc77717016f5dc21db102f95ff353476bf96849dff5d65cb94ef02dR1815). You can just push to my fork. If you need help, ask in the observability-docs channel.

File: docs/en/observability/aws-firehose-index.asciidoc
Comment (dedemorton) [inline docs/en/observability/aws-firehose-index.asciidoc:1]: @tommyers-elastic Here's an index file to get you started if you want to get these docs published while I'm away. You can just add this alias and reload your bash profile (or equivalent):

The command for building this locally looks like:

`$GIT_HOME/docs/build_docs --doc $GIT_HOME/observability-docs/docs/en/observability/aws-firehose-index.asciidoc --chunk 2`

File: docs/en/observability/aws-firehose/aws-firehose.asciidoc
Comment (dedemorton) [inline docs/en/observability/aws-firehose/aws-firehose.asciidoc:31]: I think this OK, but you might want to remove the nesting for this to work better in a separate book. 
image

**Comment by dedemorton** [comment]
This [commit](https://github.com/elastic/observability-docs/pull/2802/commits/d3c4fcb474eea09e64682759666cf7408c680545) should fix the build errors. The formatting in the section I fixed is a bit eccentric. I'll want revisit the formatting when I have more time to work on these docs. To make the source easier to read (for now), I've used block delimiters (`--`) rather than using the line continuation character multiple times. I've also added the anchors to the beginning of each step so they don't mess up list numbering. This should work, but in general, we don't link to steps in procedures because it's a bit clunky in our current doc system. Leaving as-is for now...we can fix this later.

**Comment by dedemorton** [comment]
@tommyers-elastic Assuming the doc build passes, can you make sure the format of the output matches what you expect? After you've done that, I want to remove the include from the index file in the observability guide (to ensure that the pages never get published there). Then I can get these docs building in the new location for "cloud native" ingestion.

**Comment by dedemorton** [comment]
@tommyers-elastic I've started a [PR to get these files building](https://github.com/elastic/docs/pull/2671) under the new container for cloud native docs. After we get #2802 finalized and merged, I should be able to get the docs building pretty quickly.

What's left to do:
- [ ] @tommyers-elastic Verify that the [preview](https://observability-docs_2802.docs-preview.app.elstc.co/guide/en/observability/master/aws-firehose.html) looks as you expect.
- [ ] @dedemorton Edit the index.asciidoc to remove these docs from the Observability Guide.
- [ ] Merge this PR.
- [ ] Rebuild and merge https://github.com/elastic/docs/pull/2671

**Comment by tommyers-elastic** [comment]
hey dede there are a couple of very small formatting issues to iron out. please drop me a message on slack later and we can quickly take a look together. thanks so much for all your help on this PR.

**Review by dedemorton — APPROVED** [review]
LGTM--assuming the build has no errors. I'll merge this when it goes green.

**Pull request merged**
